src/autopas/containers/verletClusterLists/VerletClusterListsRebuilder.h

```diff
@@ -240,45 +241,54 @@ class VerletClusterListsRebuilder {
+   * @param towerIndexY The y index of the given tower.
+   * @param minNeighborIndexX The minimum neighbor tower index in x direction.
+   * @param maxNeighborIndexX The maximum neighbor tower index in x direction.
+   * @param minNeighborIndexY The minimum neighbor tower index in y direction.
+   * @param maxNeighborIndexY The maximum neighbor tower index in y direction.
+   * @param useNewton3 Specifies, whether neighbor lists should contain only forward neighbors.
+   * @param function Function to apply on every neighbor tower. Typically this is calculateNeighborsBetweenTowers().
    */
-  void calculateNeighborsForTowerInRange(const int towerIndexX, const int towerIndexY, const int minX, const int maxX,
-                                         const int minY, const int maxY, const bool useNewton3) {
-    auto &tower = getTowerAtCoordinates(towerIndexX, towerIndexY);
+  template <class FunType>
+  void iterateNeighborTowers(const int towerIndexX, const int towerIndexY, const int minNeighborIndexX,
+                             const int maxNeighborIndexX, const int minNeighborIndexY, const int maxNeighborIndexY,
+                             const bool useNewton3, FunType function) {
+    auto &tower = getTower(towerIndexX, towerIndexY);
     // for all neighbor towers
-    for (int neighborIndexY = minY; neighborIndexY <= maxY; neighborIndexY++) {
+    for (int neighborIndexY = minNeighborIndexY; neighborIndexY <= maxNeighborIndexY; neighborIndexY++) {
       double distBetweenTowersY = std::max(0, std::abs(towerIndexY - neighborIndexY) - 1) * _towerSideLength;
 
-      for (int neighborIndexX = minX; neighborIndexX <= maxX; neighborIndexX++) {
-        if (useNewton3 and not shouldTowerContainOtherAsNeighborWithNewton3(towerIndexX, towerIndexY, neighborIndexX,
-                                                                            neighborIndexY)) {
+      for (int neighborIndexX = minNeighborIndexX; neighborIndexX <= maxNeighborIndexX; neighborIndexX++) {
+        if (useNewton3 and not isForwardNeighbor(towerIndexX, towerIndexY, neighborIndexX, neighborIndexY)) {
           continue;
         }
 
         double distBetweenTowersX = std::max(0, std::abs(towerIndexX - neighborIndexX) - 1) * _towerSideLength;
 
-        // calculate distance in xy-plane and skip if already longer than interactionLength
+        // calculate distance in xy-plane
         auto distBetweenTowersXYsqr = distBetweenTowersX * distBetweenTowersX + distBetweenTowersY * distBetweenTowersY;
+        // skip if already longer than interactionLength
         if (distBetweenTowersXYsqr <= _interactionLengthSqr) {
-          auto &neighborTower = getTowerAtCoordinates(neighborIndexX, neighborIndexY);
+          auto &neighborTower = getTower(neighborIndexX, neighborIndexY);
 
-          calculateNeighborsForTowerPair(tower, neighborTower, distBetweenTowersXYsqr, useNewton3);
+          (this->*function)(tower, neighborTower, distBetweenTowersXYsqr, useNewton3);
```

Comment:
I don't like this method of passing member functions.
Best solution would be to make the passed function static.